core/kernel/console.c

```diff
@@ -129,10 +129,8 @@ void configure_console_from_dt(void)
 
 	fdt = get_dt();
 
-#if defined(CFG_CBMEM_CONSOLE)
-	if (cbmem_console_init_from_dt(fdt))
+	if (IS_ENABLED(CFG_CBMEM_CONSOLE) && cbmem_console_init_from_dt(fdt))
```

Comment:
This fixup requires there is an inline definition for `cbmem_console_init_from_dt()` in _cbmem_console.h_ when `CFG_CBMEM_CONSOLE` is disabled, which means the inline definition could return false is that case and `IS_ENABLED()` condition above removed.

Reply:
I thought that too before I tried to compile it. Apparently the compiler optimizes it and doesn't even try to build the rest of the if the 'if' statement since the first part of the boolean is false. So this does actually work.

Reply:
all fine. I tested with old 32b and 64b toolchains, they're all happy.
Discard my comment.